nipype/pipeline/engine/tests/test_newnode_neuro.py

```diff
@@ -0,0 +1,120 @@
+from ..auxiliary import Function_Interface
+
+#dj niworkflows vs ...??
+from niworkflows.nipype.interfaces.utility import Rename
+import niworkflows.nipype.interfaces.freesurfer as fs
+
+from ...interfaces.freesurfer import PatchedConcatenateLTA as ConcatenateLTA
+
+Name = "example"
+DEFAULT_MEMORY_MIN_GB = None
+Inputs = {}
+
+# wf = Workflow(name, mem_gb_node=DEFAULT_MEMORY_MIN_GB,
+#               inputs=['source_file', 't1_preproc', 'subject_id',
+#                       'subjects_dir', 't1_2_fsnative_forward_transform',
+#                       'mem_gb', 'output_spaces', 'medial_surface_nan'],
+#               outputs='surfaces')
+#
+#dj: why do I need outputs?
+wf = NewWorkflow(name=Name, mem_gb_node=DEFAULT_MEMORY_MIN_GB, inputs=Inputs)
+
+
+# @interface
+# def select_target(subject_id, space):
+#     """ Given a source subject ID and a target space, get the target subject ID """
+#     return subject_id if space == 'fsnative' else space
+
+def select_target(subject_id, space):
+    """ Given a source subject ID and a target space, get the target subject ID """
+    return subject_id if space == 'fsnative' else space
+
+#select_target_interface = Function_Interface(select_target, ["out"])
+
+
+# wf.add('targets', select_target(subject_id=wf.inputs.subject_id))
+#   .map('space', space=[space for space in wf.inputs.output_spaces
+#                        if space.startswith('fs')])
+
+#dj: don't have option in map to connect with wf input
+
+wf.add(runnable=select_target, name="targets", subject_id="subject_id")\
+    .map(mapper="space", inputs={"space": [space for space in Inputs["output_spaces"]
+                                           if space.startswith("fs")]})
+
+
+# wf.add('rename_src', Rename(format_string='%(subject)s',
+#                             keep_ext=True,
+#                             in_file=wf.inputs.source_file))
+#   .map('subject')
+
+
+# dj TODO: how do we get subject? how do we use it in the Rename?
+wf.add(name='rename_src',
+       runnable=Rename(format_string='%(subject)s', keep_ext=True),
+                            in_file="source_file")\
+    .map('subject')
+
+
+# wf.add('resampling_xfm',
+#        fs.utils.LTAConvert(in_lta='identity.nofile',
+#                            out_lta=True,
+#                            source_file=wf.inputs.source_file,
+#                            target_file=wf.inputs.t1_preproc)
+#   .add('set_xfm_source', ConcatenateLTA(out_type='RAS2RAS',
+#                                         in_lta2=wf.inputs.t1_2_fsnative_forward_transform,
+#                                         in_lta1=wf.resampling_xfm.out_lta))
+
+
+wf.add(name='resampling_xfm',
+       runnable=fs.utils.LTAConvert(in_lta='identity.nofile', out_lta=True),
+       source_file="source_file", target_file="t1_preproc")\
+    .add(name='set_xfm_source', runnable=ConcatenateLTA(out_type='RAS2RAS'),
+         in_lta2="t1_2_fsnative_forward_transform", in_lta1="resampling_xfm.out_lta")
```

Comment:
how does this node know, that the `t1_2_fsnative_forward_transform` is a workflow input?

Reply:
for node output I uses node.output, e.g., `in_lta1="resampling_xfm.out_lta"`